### DISCUSSION TITLE
L3, R3, Select, Start, and Pressure Buttons do not work

### DISCUSSION THREAD

**Discussion by MysticAx0lotl** [nefarius/DsHidMini#230] [Support]
Hello there!
For some reason, in PCSX2 1.7.3103, I cannot get L3, R3, Select, and Start to recognize as inputs, and I cannot get the emulator to properly detect the pressure-sensitive face buttons. I've tried changing input types, I've tried both Bluetooth and USB, and I've tried mapping with "mute digital inputs" both on and off. What am I doing wrong here? Thanks!

(also, sorry for posting this in the wrong place initially, I keep forgetting to check discussions before I submit an issue)

**Comment by nefarius** [comment]
What mode are you using? Have you tried the SCP XInput DLL variant?

**Reply by MysticAx0lotl** [reply]
No, this is a completely fresh install of PCSX2, with no mods as far as I'm aware. I uninstalled SCP and rebooted my machine before I set this instance up.

**Reply by nefarius** [reply]
[Put this DLL](https://buildbot.vigem.org/builds/DsHidMini/master/3.0.371.0/bin/x64/XInput1_3.dll) in the installation directory of PCSX2, switch to SXS mode and in PCSX2 configure the pad using the XInput checkbox.

**Reply by MysticAx0lotl** [reply]
I tried resetting my controller (which I didn't even know was an option until 30 seconds after I made this issue, despite owning a PS3 since 2011), and the same issues were present.

**Reply by MysticAx0lotl** [reply]
> [Put this DLL](https://buildbot.vigem.org/builds/DsHidMini/master/3.0.371.0/bin/x64/XInput1_3.dll) in the installation directory of PCSX2, switch to SXS mode and in PCSX2 configure the pad using the XInput checkbox.

Trying this now.

**Reply by nefarius** [reply]
[I made a video how to do it](https://vigem.org/projects/DsHidMini/SCP-XInput-Bridge/), it's for GTA SA but the concept is the same.

**Reply by MysticAx0lotl** [reply]
It still doesn't work. I'm going to try rebooting my PC again, and see if that helps.

**Reply by MysticAx0lotl** [reply]
Same issue, unfortunately.

**Reply by nefarius** [reply]
Well, I can guarantee you this method works. Tested any other games? Or gamepad tester website? Made sure the hardware is good?

**Reply by MysticAx0lotl** [reply]
> Well, I can guarantee you this method works. Tested any other games? Or gamepad tester website? Made sure the hardware is good?

Hardware is fine. I tested on my PS3, on the SDL Gamepad Tester, and Sonic Generations, and all inputs worked correctly. Maybe this could offer some insight? This is a screenshot from PCSX2:
![image](https://user-images.githubusercontent.com/69944483/180102851-cdc2a95a-a3bc-4a76-aec4-1c1ab12f04dc.png)

**Reply by nefarius** [reply]
Show the entire settings screen. Is the DLL in the correct directory?

**Reply by MysticAx0lotl** [reply]
![image](https://user-images.githubusercontent.com/69944483/180104050-36682cee-4f5f-46ee-a991-3445e94dfd7d.png)
![image](https://user-images.githubusercontent.com/69944483/180104072-7bb05e49-42de-40b6-a52a-de28745e483a.png)
![image](https://user-images.githubusercontent.com/69944483/180104093-b72214fd-571d-42a2-99b1-0183a38a7ef9.png)

**Reply by nefarius** [reply]
Disable SDL. You need to use XInput only. Did you redo the mapping afterwards?

**Reply by MysticAx0lotl** [reply]
The buttons still don't work. Also, sometimes this happens (I only have 1 controller hooked up via USB):
![image](https://user-images.githubusercontent.com/69944483/180104340-f5a96463-217b-41cc-a1a7-08a3bde2d793.png)

**Reply by nefarius** [reply]
Welp, IDK then.

**Reply by nefarius** [reply]
Ha

**Reply by nefarius** [reply]
Are you in SXS mode? Show settings.

**Reply by nefarius** [reply]
> The buttons still don't work. Also, sometimes this happens (I only have 1 controller hooked up via USB): ![image](https://user-images.githubusercontent.com/69944483/180104340-f5a96463-217b-41cc-a1a7-08a3bde2d793.png)

Disconnect all, start over.

**Reply by MysticAx0lotl** [reply]
I'm out of time tonight, I'm going to try uninstalling and reinstalling everything tomorrow. Hopefully that helps.

**Reply by MysticAx0lotl** [reply]
@nefarius I got it connected, and I got L3, R3, Start, and Select to map, but the pressure-sensitivity doesn't work at all. I ended up having to set DsHidMini to Xinput mode, and enabling SDL in PCSX2. Edit: This is after a complete reinstall of everything.

**Reply by nefarius** [reply]
Pressure sensitivity only works in SDF mode or in SXS mode in combination with the XInput proxy DLL.

**Reply by MysticAx0lotl** [reply]
> Pressure sensitivity only works in SDF mode or in SXS mode in combination with the XInput proxy DLL.

The problem is when in SDF mode, I get the issues of certian buttons not mapping. I haven't tried SXS recently though.

**Reply by MysticAx0lotl** [reply]
SXS still doesn't work with pressure-sensitivity, even with the Xinput DLL

**Reply by MysticAx0lotl** [reply]
Nevermind. I have no clue how this fixed it (as I've tried it before), but I 1. Set DsHidMini into SXS mode, 2. Re-installed the custom XInput DLL (IDK if this did anything), 3. Disabled SDL and enabled Xinput in PCSX2, and it worked. Thank you for all your help!
